src/sphinxfilter.cpp

```diff
@@ -2313,8 +2328,22 @@ void FormatFilterQL ( const CSphFilterSettings & f, StringBuilder_c & tBuf, int
 
 		case SPH_FILTER_USERVAR:
 		case SPH_FILTER_STRING:
-			tBuf.Sprintf ( "%s%s'%s'", f.m_sAttrName.cstr(), ( f.m_bExclude ? "!=" : "=" ), ( f.m_dStrings.GetLength()==1 ? f.m_dStrings[0].scstr() : "" ) );
+		{
+			const char * op = ([&]() {
```

Comment:
we use `sOp` convention described at [internals-coding-standard.txt](https://github.com/manticoresoftware/manticoresearch/blob/master/doc/internals-coding-standard.txt) 
that is not critical code and it could be as you wish but why use lambda instead of the plain switch?
```
const char * sOp = "=";
switch ( f.m_eStrCmpDir )
				{
					case EStrCmpDir::NE: sOp = "!="; break;
					...
					default:
							sOp = "="; break;
```

Reply:
This pattern avoids having to reassign the variable (due to an essentially useless default `"="`), and it also avoids the extra `break;` statement for each case. I guess avoiding variable reassignment is just a habit carried over from functional languages, where variables can be assigned to only once, but imo it makes code easier to reason about when variables receive their values in exactly one location.

And sure, I can rename it to `sOp`.